pkg/admission/admission.go

```diff
@@ -154,6 +154,23 @@ func (a *ParallelAdmission) ValidateResources(ctx context.Context, localResource
 			Name:      objName,
 		}
 
+		// If the object includes a pod spec (i.e. Deployment), create a Pod object out of it for validation.
+		// For admission only pod resources will be enforced, deployments won't.
+		podExtractor := psadmission.DefaultPodSpecExtractor{}
```

Comment:
A small refactoring proposal:
Would you make the `PodExtractor` an attribute of the `ParallelAdmission` struct, create and assign it in `NewParallelAdmission()` and also pass it as an argument to `setupAdmission()` in the same function?

This should help in case anyone wanted to use this as a library and supply their own `PodExtractor` some time in the future.